geode-management/src/main/java/org/apache/geode/management/configuration/Region.java

```diff
@@ -271,19 +268,27 @@ public void setAction(EvictionAction action) {
     }
 
     public Integer getEntryCount() {
-      return entryCount;
+      if (type == EvictionType.ENTRY_COUNT) {
+        return limit;
+      }
+      return null;
     }
 
     public void setEntryCount(Integer entryCount) {
-      this.entryCount = entryCount;
+      type = EvictionType.ENTRY_COUNT;
```

Comment:
I think I liked the way this was before this change. Now if a user sets both entryCount and memorySizeMb then the last one wins. If a users thinks they can set both (they might think we support both an entry count limit AND a memory size limit) then they would set both. Before this would have given them an error but now it just switches to the last one. And if they are using JSON then how do they even know which one is last? If they do this {entryCount=10; memorySizeInMb=100} they will not get an error anymore but instead it will be either one or the other. And if they also set the type to HEAP_PERCENTAGE then this may also get reset if these settors are invoked later. Before we just had each settor setting one attribute which I think is clearer to users.

Reply:
Yes, that is an issue with this approach.

Reply:
My issue with the previous approach is that it gives user too many opportunities to create an invalid eviction and it relies heavily on the validator to catch all the invalid cases.

Reply:
I pushed in another commit to validate type been changed by setter.  Please check if that addressed Darrel's concern.

Reply:
> My issue with the previous approach is that it gives user too many opportunities to create an invalid eviction and it relies heavily on the validator to catch all the invalid cases.

Given the JSON that a user can submit that does not really change with either approach. I think the point of the validator is that it does exactly that; catch the cases in which the user submits invalid requests.

Reply:
I think the differences is the later approach would give user less opportunities to create an invalid object to begin with, whereas the the previous approach will let user create any combination of attributes and let validator to catch all the invalid cases.

Reply:
I like the changes you have made. It seems like the constructors on Eviction still allow for just as many illegal creations to be done. The settors now do more validation but the constructor that takes all the attributes does not do any validation. I'm not sure if it matters to the user if the validation happens on a method/constructor call vs. when they call ClusterManagementService.create.